src/main/java/org/candlepin/subscriptions/db/model/Subscription.java

```diff
@@ -83,4 +84,14 @@ public SubscriptionCompoundId() {
       // default
     }
   }
+
+  public void endSubscription() {
+    endDate = OffsetDateTime.now();
+  }
+
+  public boolean quantityHasChanged(long newQuantity) {
+    return this.getQuantity() != newQuantity;
+  }
+
+  // TODO: Add builder to throw exceptions if required fields are missing //NOSONAR
```

Comment:
What's up with this TODO that was also tagged with `//NOSONAR`?  Is this something that should be completed as part of ENT-2659 or is there another story that this should be part of?  If it's the latter, I'd rather see this TODO as part of the description in a JIRA card that addresses it instead of a comment here.

cc: @kahowell

Reply:
+1, I would add it's nice to put the Jira reference here if you do end up pushing out to a backlog item.

Reply:
Added subtasks to ENT-2659 as discussed. I need to add the //NOSONAR tag to avoid sonarqube check failures.